modules/data.atmosphere/R/tdm_temporal_downscale_functions.R

```diff
@@ -0,0 +1,1127 @@
+        # window
+        if (is.na(summary(mod.doy)$adj.r.squared)) {
+            stop(paste0("Can not estimate covariance matrix for day of year: ", 
+                unique(dat.subset$doy), ";  Increase day.window and try again"))
+            # dat.subset <- rbind(dat.subset, dat.subset) mod.doy <- lm(rain.prop ~
+            # as.factor(hour)*precipitation_flux.day-1 -
+            # as.factor(hour)-precipitation_flux.day, data=dat.subset)
+        }
+        
+        # Generate a bunch of random coefficients that we can pull from without
+        # needing to do this step every day
+        mod.coef <- coef(mod.doy)
+        mod.cov <- vcov(mod.doy)
+        piv <- as.numeric(which(!is.na(mod.coef)))
+        Rbeta <- MASS::mvrnorm(n = n.beta, mod.coef[piv], mod.cov)
+        
+        list.out <- list(model = mod.doy, betas = Rbeta)
+        # Model residuals as a function of hour so we can increase our
+        # uncertainty
+        if (resids == T) {
+            # dat.subset[!is.na(dat.subset$lag.precipitation_flux) &
+            # !is.na(dat.subset$next.precipitation_flux),'resid'] <- resid(mod.doy)
+            dat.subset[, "resid"] <- resid(mod.doy)
+            resid.model <- lm(resid ~ as.factor(hour) * precipitation_flux.day - 
+                1, data = dat.subset[, ])
+            res.coef <- coef(resid.model)
+            res.cov <- vcov(resid.model)
+            res.piv <- as.numeric(which(!is.na(res.coef)))
+            
+            beta.resid <- MASS::mvrnorm(n = n.beta, res.coef[res.piv], 
+                res.cov)
+            
+            list.out[["model.resid"]] <- resid.model
+            list.out[["betas.resid"]] <- beta.resid
+        }
+        
+        return(list.out)
+    }
+    
+    dat.list <- list()
+    mod.out <- list()
+    
+    # Make the data into a list Training the model on ax X-day window
+    # around the actual DOY we're trying to model this helps avoid problems
+    # with lack of data in small datasets like Ameriflux Default window is
+    # 5 days (+/- 2)
+    for (i in unique(dat.train$doy)) {
+        if (i >= 365) {
+            # Lump leap day in with non-leap Dec 31
+            dat.list[[paste(i)]] <- dat.train[dat.train$doy >= 365 - day.window/2 | 
+                dat.train$doy <= day.window/2, ]
+        } else if (i == 1) {
+            dat.list[[paste(i)]] <- dat.train[dat.train$doy <= i + day.window/2 | 
+                dat.train$doy >= 365 - day.window/2, ]
+        } else {
+            dat.list[[paste(i)]] <- dat.train[dat.train$doy >= i - day.window/2 & 
+                dat.train$doy <= i + day.window/2, ]
+        }
+    }
+    
+    # Do the computation and save a list Final list will have 2 layers per
+    # DOY: the model, and a bunch of simulated betas
+    if (parallel == T) {
+        warning("Running model calculation in parallel.  This WILL crash if you do not have access to a LOT of memory!")
+        library(parallel)
```

Comment:
according to our coding style ... use `TRUE` instead of `T`

recognizing that `parallel == TRUE`, `(parallel == TRUE) == TRUE`,  `((parallel == TRUE) == TRUE) == TRUE`, `(parallel == TRUE) == (parallel == TRUE)` etc (I was having fun, sorry it got out of hand), 

it should be sufficient to use

```r
if(parallel){
   ...
}
```